docs/introduction-to-smart-contracts.rst

```diff
@@ -563,9 +563,10 @@ idea, but it is potentially dangerous, as if someone sends Ether to removed
 contracts, the Ether is forever lost.
 
 .. warning::
-    From version 0.8.18 and up, the use of ``selfdestruct`` in both Solidity and Yul will trigger a
-    deprecation warning, since the ``SELFDESTRUCT`` opcode will eventually undergo breaking changes in behavior
-    as stated in `EIP-6049 <https://eips.ethereum.org/EIPS/eip-6049>`_.
+    From ``EVM >= Cancun`` onwards, ``selfdestruct`` will **only** send all Ether in the account to the given recipient and not destroy the contract. However, when ``selfdestruct`` is called in the same transaction as the contract calling ``selfdestruct`` was created, the behaviour of ``selfdestruct`` pre-Cancun hardfork (i.e., ``EVM <= Shanghai``) is preserved and will destroy the current contract, deleting any data, including storage keys, code and the account itself. See `EIP-6780 <https://eips.ethereum.org/EIPS/eip-6780>`_ for more details.
+
```

Comment:
One thing that's still missing is https://github.com/ethereum/solidity/pull/14803#discussion_r1511340392. I.e. I think it's important to mention the distinction between EVM version used by the network and the EVM version we're compiling for.

We could use the wording from the blog post:
```
This is a network-wide change that will affect all deployed contracts.
The ``--evm-version`` setting used when compiling the contract has no bearing on it.
```

Reply:
I see you added this to one of the other warnings. I think it's related to the one here instead so I just pushed a commit that moved it here.